src/NServiceBus.Core/OpenTelemetry/Tracing/ActivityFactory.cs

```diff
@@ -54,9 +66,22 @@ public Activity StartIncomingActivity(MessageContext context)
         return activity;
     }
 
+    /// <summary>
+    /// Message can be delayed due to requesting a saga timeout, deferring a message through send options or delayed retries.
+    /// Saga timeout and message deferral will result in the DeliverAt header set
+    /// Delayed retry will result in the DelayedRetries header set
+    /// </summary>
+    /// <param name="context"></param>
+    /// <returns></returns>
+    static bool IsMessageDelayed(MessageContext context) =>
+        context.Headers.ContainsKey(Headers.DeliverAt) ||
+        context.Headers.ContainsKey(Headers.DelayedRetries);
+
+    // This could be cleaned up once a dedicated API is created, see https://github.com/dotnet/runtime/issues/65528
```

Comment:
```suggestion
    /// <summary>
    /// This could be cleaned up once a dedicated API is created, see https://github.com/dotnet/runtime/issues/65528
    /// </summary>
```